### PR TITLE
refactor: make monitor test template more compact

### PR DESCRIPTION
**Pull request by dragonly**
### What problem does this PR solve?

refactor monitor test template to make it more compact.

### What is changed and how does it work?

use template to extract common parts

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [x] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
None
```

### CODE REVIEW

File: pkg/monitor/monitor/template.go [hunk 1]
```diff
@@ -366,11 +359,10 @@ func scrapeJob(jobName string, componentPattern config.Regexp, cmodel *MonitorCo
...
 				{
 					SourceLabels: model.LabelNames{
-						metricsPathLabel,
+						componentLabel,
 					},
-					Action:      config.RelabelReplace,
-					TargetLabel: "__metrics_path__",
-					Regex:       allMatchPattern,
+					Action: config.RelabelKeep,
+					Regex:  componentPattern,
```
Comment (DanielZhangQD) [inline pkg/monitor/monitor/template.go:365]: Why change this block?

File: pkg/monitor/monitor/template_test.go [hunk 2]
```diff
@@ -682,12 +355,19 @@ remote_write:
...
 	content, err := RenderPrometheusConfig(model)
 	g.Expect(err).NotTo(HaveOccurred())
-	g.Expect(content).Should(Equal(expectedContent))
+	expectedContentParsed := template.Must(template.New("relabelConfig").Parse(expectedContentTpl))
+	var expectedContentBytes bytes.Buffer
+	expectedContentParsed.Execute(&expectedContentBytes, promCfgModel)
+	// fmt.Print(len(content))
+	// fmt.Println("==============")
+	// fmt.Println(expectedContentBytes.String())
+	// fmt.Print(len(expectedContentBytes.String()))
```
Comment (DanielZhangQD) [inline pkg/monitor/monitor/template_test.go:364]: Should this be removed?

Comment (dragonly) [reply pkg/monitor/monitor/template_test.go:364]: yes, my bad

File: pkg/monitor/monitor/template.go (hunk 1 above)
Comment (dragonly) [reply pkg/monitor/monitor/template.go:365]: It seems the github diff issue

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Pull request merged**